core/src/main/java/io/methvin/watcher/changeset/ChangeSetListener.java

```diff
@@ -13,7 +18,25 @@
 
   private final Object lock = new Object() {};
 
-  public ChangeSetListener() {}
+  private final int timeout;
+
+  private ScheduledExecutorService service;
```

Comment:
nit: can be final

Reply:
ScheduledExecutorService is setted only if we need OnIdle